File: 第一次课/第1题/t1.py
# -*- coding: utf-8 -*-

import numpy as np

N=100000


x=[]
for k in range(0,N):
    x.append(1/N*k)


dx=[]#使用空数组
y=[]

for k in range(0,N-1):
    y.append(np.exp(x[k]))
    dx.append(x[k+1]-x[k])

x=np.matrix(x)#矩阵化
y=np.matrix(y)
dx=np.matrix(dx)


I_num=np.sum(np.multiply(y,dx))
#I_num=np.dot(y,dx.transpose(),)[0,0]

I_exact=np.e-1

error=np.abs(I_exact-I_num)

print('I_exact:',I_exact)
print('I_num:',I_num)
print('error:',error)

File: 第一次课/第1题/t1.m
clear

N=100000;               %设定分割量
x=0:1/N:1;              %在区间上建立分割点，从0开始
dx=zeros(N,1);          %dx值存储向
y=zeros(N,1);           %y值存储向量。

for k=1:N 
   y(k)=exp(x(k));      %对应y值
   dx(k)=x(k+1)-x(k);   %对应dx
end

I=sum(y.*dx);           %内积的方式求和

I_exact=exp(1)-1        %计算真实值
I_num=I                 %保留计算值
error=abs(I_exact-I_num)%计算绝对误差限




